src/metagen/src/client_ts/static/mod.ts

```diff
@@ -436,22 +543,43 @@ async function fetchGql(
   doc: string,
   variables: Record<string, unknown>,
   options: GraphQlTransportOptions,
+  files?: Map<string, File>,
 ) {
-  // console.log(doc, variables);
+  let body: FormData | string = JSON.stringify({
+    query: doc,
+    variables,
+  });
+
+  const additionalHeaders: HeadersInit = {};
+
+  if (files && files.size > 0) {
+    const data = new FormData();
+    data.set("operations", body);
+    const map: Record<string, string[]> = {};
+    for (const [i, [path, file]] of [...(files?.entries() ?? [])].entries()) {
+      const key = `${i}`;
+      // TODO single file on multiple paths
```

Comment:
By the way @Natoandro, what do you exactly mean by this?

Reply:
If the same file is used multiple times, then it should not be emitted more than once in the request payload... Should be added to the array...

Reply:
Ah, I understand now why you used IDs for the Rust implementation.

Reply:
I'll make a follow up PR for the python one and add more tests for the path extraction